collider.lua

```diff
@@ -45,6 +45,12 @@ end
 
 
 function Collider:destroy()
+   for i, v in ipairs(self._world.colliders) do
```

Comment:
Oh I see, when I changed the lookup table to a list in sort-colliders I broke everything that tries to find a collider in world.colliders. A consequence of poor encapsulation on my part.

This approach works but is terribly inefficient and doesn't fix the encapsulation issue. For the encapsulation I'd suggest a function `World:_remove(collider)` (underscore because the user of the library shouldn't use it, they may otherwise think it destroys the collidert) . I can't think of a way that is more efficient (everything that comes to mind means an equally expensive operation whenever the draw order changes. So, world:_remove(collider) would probably contain the loop that you've written here. This would reduce the risk of similar issues cropping up elsewhere.

Reply:
`World:_remove` successfully added :+1:

It may be possible to make the array removal more efficient by using sorted table of some kind, but I am not that experienced in Lua to implement such a data structure.